src/v/storage/ntp_config.h

```diff
@@ -376,19 +380,18 @@ class ntp_config {
 
     std::unique_ptr<default_overrides> _overrides;
 
-    /**
-     * A number indicating an id of the NTP in case it was created more
-     * than once (i.e. created, deleted and then created again)
-     */
+    /// Revision of the command that resulted in this partition appearing on
+    /// this node (i.e. it will changed if the partition replica is moved back
+    /// and forth from/to the node, as well as if the topic is re-created). It
+    /// is used in constructing the local directory path.
     model::revision_id _revision_id{0};
 
-    /**
-     * A number indicating an initial revision of the NTP. The revision
-     * of the NTP might change when the partition is moved between the
-     * nodes. The initial revision is the revision_id that was assigned
-     * to the topic when it was created.
-     */
-    model::initial_revision_id _initial_rev{0};
+    /// Revision of the topic creation command.
+    model::revision_id _topic_rev;
```

Comment:
> nit: should this have a default value like the others?
0 is a valid revision id, unlikely that a topic creation command will be at offset 0, but still... for `_revision_id` it kind of makes sense to have a default of 0 (for tests and such), so that we don't have directories with names like `data/kafka/foo/0_-gazillion`, but for `_topic_rev` I'm not sure.

> nit: maybe this should be yet another named type for topic revision?

Not a big fan of a different named type for this tbh. Danger of mixing seems low, and we'll have to cast back and forth anyway in many places.